shared/src/replicate.rs

```diff
@@ -14,8 +14,48 @@ pub trait Replicate<P: ProtocolType>: ReplicateSafe<P> {
     /// Returns a clone of self
     fn clone(&self) -> Self;
 }
+#[cfg(feature = "bevy")]
+use bevy::{ecs::component::TableStorage, prelude::Component};
 
-/// The part of ReplicateSafe which is object-safe
+/// The part of ReplicateSafe which is object-safe and has bevy support
```

Comment:
This can be:

```rust
#[cfg(feature = "bevy")]
impl<P: ProtocolType> Component for ReplicateSafe<P> {
   type Storage = TableStorage;
}
```

And you can drop the duplicate trait definition.

Reply:
that gives me a warning that it needs to have "for dyn ReplicateSafe<P>" because its deprecated, and it also refuses to compile when using that. 

Tried many things, after a while, that was the one I landed on..

Reply:
Ah, my bad. I was thinking something like:

```rust
impl<P: ProtocolType, R: ReplicateSafe<P>> Component for R {
    type Storage = TableStorage;
}
```
but that isn't actually possible to specify. The macro seems like a good approach.